### PR TITLE
[profiler] Add duration=NUM parameter to AOT profiler

### PR DESCRIPTION
**Pull request by radekdoulik**
It instructs the AOT profiler to write the data after NUM seconds
measured from runtime initialization.

Fixes: https://github.com/xamarin/xamarin-android/issues/2813

### CODE REVIEW

**Comment by marek-safar** [comment]
@lambdageek please review

File: mono/profiler/aot.c [hunk 1]
```diff
@@ -34,6 +36,8 @@ struct _MonoProfiler {
 	char *outfile_name;
 	mono_mutex_t mutex;
 	gboolean verbose;
+	MonoNativeThreadId helper_thread;
```
Comment (lambdageek) [inline mono/profiler/aot.c]: this field seems unused.

**Comment by vargaz** [comment]
I think the proper way to fix this is to have some kind of api to control the profiler.

**Comment by radekdoulik** [comment]
I though about various way to achieve this, in the end that was the simplest way for us as we can just adjust the arguments. (we set it by `adb setprop` from command line)

For later usage from UI I was thinking about listening on a socket, similar to what log profiler does.

**Comment by radekdoulik** [comment]
@monojenkins backport to 2018-10

**Comment by radekdoulik** [comment]
@monojenkins backport to 2019-02

**Pull request merged**